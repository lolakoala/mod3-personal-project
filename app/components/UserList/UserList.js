import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BillsList from '../BillsList/BillsList.js';
import ChoresList from '../ChoresList/ChoresList.js';

class UserList extends Component {
  constructor() {
    super();
    this.state = {
      currentView: ''
    };
  }

  isActive = button => {
    return this.state.currentView === button ? 'active' : '';
  }

  render() {
    const {
      usersHouse,
      currentUser,
      markBillPaid,
      match,
      markChoreDone
    } = this.props;
    const placeRendered = match.url;
    return (
      <div className='user-list'>
        <button
          className={this.isActive('bills')}
          onClick={() => { this.setState({ currentView: 'bills'}); }}>
          View Bills
        </button>
        <button
          className={this.isActive('chores')}
          onClick={() => { this.setState({ currentView: 'chores'}); }}>
          View Chores
        </button>
        {this.state.currentView === 'bills' ?
          <BillsList usersHouse={usersHouse}
            currentUser={currentUser}
            placeRendered={placeRendered}
            markBillPaid={markBillPaid}/>
          : null}
        {this.state.currentView === 'chores' ?
          <ChoresList usersHouse={usersHouse}
            currentUser={currentUser}
            placeRendered={placeRendered}
            markChoreDone={markChoreDone}
            claimChore={() => {}}/>
          : null}
      </div>
    );
  }
}

export default UserList;

UserList.propTypes = {
  usersHouse: PropTypes.object,
  currentUser: PropTypes.object,
  markBillPaid: PropTypes.func,
  match: PropTypes.object,
  markChoreDone: PropTypes.func
};
