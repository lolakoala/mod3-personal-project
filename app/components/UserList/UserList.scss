.user-list {
  font-family: 'BenchNine', sans-serif;
  margin: 20px;
}

.user-list button {
  font-family: 'BenchNine', sans-serif;
  border: 2px solid #D83D59;
  background-color: #D83D59;
  width: 120px;
  height: 25px;
  color: #F4CCE2;
  font-size: 20px;
  border-radius: 25px;
  text-align: center;
  margin: 5px;
}

.user-list button:hover,
.user-list .active {
  cursor: pointer;
  border: 2px dashed #D83D59;
  background-color: #F4CCE2;
  color: #D83D59;
}
