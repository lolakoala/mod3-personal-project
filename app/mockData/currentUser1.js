const currentUser1 = {
  id : "RVde1m9BhkQBLZlTJNsFlQZXqP43",
  name: "Amanda Brenner",
  house: "Dog9"
};

export default currentUser1;
