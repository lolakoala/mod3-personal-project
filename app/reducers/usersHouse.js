import firebase from '../firebase.js';

const usersHouse = (state = {}, action) => {
  const housesRef = firebase.database().ref('houses');
  switch (action.type) {
  case 'CREATE_HOUSE':
    housesRef.push(action.house);
    return action.house;
  case 'GET_HOUSE':
    const updatedHouse = Object.assign(action.usersHouse, { users: [...action.usersHouse.users, action.user] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(updatedHouse);
    return updatedHouse;
  case 'LOGIN_SUCCESS':
    return action.usersHouse;
  case 'SIGNOUT':
    return {};
  case 'LEAVE_HOUSE':
    const deleteUser = action.usersHouse.users.filter(user => user.id !== action.currentUser.id);
    const houseWithoutUser = Object.assign(action.usersHouse, { users: deleteUser });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithoutUser);
    return {};
  case 'ADD_BILL':
    const takeOutFakeBills = action.usersHouse.bills.filter(bill => bill.title !== 'fake');
    const houseWithBill = Object.assign(action.usersHouse, { bills: [...takeOutFakeBills, action.bill] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithBill);
    return houseWithBill;
  case 'ADD_BULLETIN':
    const takeOutFakeBulletins = action.usersHouse.bulletins.filter(bulletin => bulletin.title !== 'fake');
    const houseWithBulletin = Object.assign(action.usersHouse, { bulletins: [...takeOutFakeBulletins, action.bulletin] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithBulletin);
    return houseWithBulletin;
  case 'ADD_CHORE':
    const takeOutFakeChore = action.usersHouse.chores.filter(chore => chore.title !== 'fake');
    const houseWithChore = Object.assign(action.usersHouse, { chores: [...takeOutFakeChore, action.chore] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(houseWithChore);
    return houseWithChore;
  case 'ADD_READER':
    const oldBulletin = action.usersHouse.bulletins.find(bulletin => bulletin.id === action.bulletinId);
    if (oldBulletin.hasRead.includes(action.userId)) {
      return action.usersHouse;
    }
    const newBulletin = Object.assign(oldBulletin, { hasRead: [...oldBulletin.hasRead, action.userId] });
    const updateBulletins = action.usersHouse.bulletins.filter(bulletin => {
      return bulletin.id !== action.bulletinId;
    });
    const updateWithBulletin = Object.assign(action.usersHouse, { bulletins: [...updateBulletins, newBulletin] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(updateWithBulletin);
    return updateWithBulletin;
  case 'MARK_PAID':
    const oldBill = action.usersHouse.bills.find(bill => bill.id === action.billId);
    const user = oldBill.allUsersTotals.find(user => user.id === action.userId);
    if (user.paid === true) {
      return action.usersHouse;
    }
    const newUser = Object.assign(user, { paid: true });
    const newAllUsersTotals = oldBill.allUsersTotals.filter(user => user.id !== action.userId);
    const newBill = Object.assign(oldBill, { allUsersTotals: [...newAllUsersTotals, newUser] });
    const updatedBills = action.usersHouse.bills.filter(bill => {
      return bill.id !== action.billId;
    });
    const updateWithBill = Object.assign(action.usersHouse, { bills: [...updatedBills, newBill] });
    firebase.database().ref("houses/" + action.usersHouse.houseKey).set(updateWithBill);
    return updateWithBill;
  default:
    return state;
  }
};

export default usersHouse;
