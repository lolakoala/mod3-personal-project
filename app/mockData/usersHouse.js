import currentUser1 from './currentUser1.js';
import currentUser2 from './currentUser2.js';

const bill1 = {
  allUsersTotals: [
    {
      id: currentUser1.id,
      paid: false,
      total: 50
    },
    {
      id: currentUser2.id,
      paid: false,
      total: 50
    }
  ],
  datePosted: "11/01/2017",
  details: "We used a lot of water.",
  duedate: "11/9/17",
  id: "b3019762705779",
  parsedDuedate: 1510210800000,
  postedBy: currentUser1,
  title: "Water Bill",
  total: "100"
};

const bulletin1 = {
  datePosted: "11/01/2017",
  details: "The dogs got out through a hole in the fence yesterday.",
  hasRead: [currentUser1.id],
  id: 1509567098747,
  postedBy: currentUser1,
  title: "Don't leave dogs out unattended."
};

const chore1 = {
  assignedTo: currentUser1.id,
  datePosted: "10/31/2017",
  details: "Empty the dishwasher.",
  done: false,
  id: "c1509485241076",
  postedBy: currentUser1,
  title: "Dishes",
  urgency: "low"
};

const usersHouse = {
  houseKey: "-KxoIBWy5UYRscFYdWty",
  bills: [bill1],
  bulletins: [bulletin1],
  chores: [chore1],
  houseCode: "woof9",
  houseName: "Dog9",
  users: [currentUser1, currentUser2]
};

export default usersHouse;
