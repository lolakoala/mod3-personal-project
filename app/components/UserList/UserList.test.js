import React from "react";
import UserList from './UserList.js';
import { shallow } from "enzyme";
import usersHouse from '../../mockData/usersHouse.js';
import currentUser1 from '../../mockData/currentUser1.js';

describe('UserList component', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<UserList usersHouse={usersHouse}
    currentUser={currentUser1}
    markBillPaid={mockFn}
    markChoreDone={mockFn}
    match={{ url: '/userlist' }}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have default state', () => {
    expect(wrapper.state().currentView).toEqual('');
  });

  it('should have a title', () => {
    const title = wrapper.find('h2');

    expect(title.text()).toEqual('My Lists');
  });

  it('should have two buttons', () => {
    const buttons = wrapper.find('button');

    expect(buttons.length).toEqual(2);
  });

  it('should change state and content when clicking view bills', () => {
    const button = wrapper.find('button').first();

    button.simulate('click');
    expect(wrapper.state().currentView).toEqual('bills');
    expect(wrapper.find('BillsList').length).toEqual(1);
  });

  it('should change state and content when clicking view chores', () => {
    const button = wrapper.find('button').last();

    button.simulate('click');
    expect(wrapper.state().currentView).toEqual('chores');
    expect(wrapper.find('ChoresList').length).toEqual(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
