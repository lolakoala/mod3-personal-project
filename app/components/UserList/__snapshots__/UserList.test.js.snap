// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserList
    currentUser={
      Object {
        "house": "Dog9",
        "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
        "name": "Amanda Brenner",
      }
    }
    markBillPaid={[Function]}
    markChoreDone={[Function]}
    match={
      Object {
        "url": "/userlist",
      }
    }
    usersHouse={
      Object {
        "bills": Array [
          Object {
            "allUsersTotals": Array [
              Object {
                "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                "paid": false,
                "total": 50,
              },
              Object {
                "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                "paid": false,
                "total": 50,
              },
            ],
            "datePosted": "11/01/2017",
            "details": "We used a lot of water.",
            "duedate": "11/9/17",
            "id": "b3019762705779",
            "parsedDuedate": 1510210800000,
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Water Bill",
            "total": "100",
          },
        ],
        "bulletins": Array [
          Object {
            "datePosted": "11/01/2017",
            "details": "The dogs got out through a hole in the fence yesterday.",
            "hasRead": Array [
              "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            ],
            "id": 1509567098747,
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Don't leave dogs out unattended.",
          },
        ],
        "chores": Array [
          Object {
            "assignedTo": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "datePosted": "10/31/2017",
            "details": "Empty the dishwasher.",
            "done": false,
            "id": "c1509485241076",
            "postedBy": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "title": "Dishes",
            "urgency": "low",
          },
        ],
        "houseCode": "woof9",
        "houseKey": "-KxoIBWy5UYRscFYdWty",
        "houseName": "Dog9",
        "users": Array [
          Object {
            "house": "Dog9",
            "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "name": "Amanda Brenner",
          },
          Object {
            "house": "Dog9",
            "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
            "name": "Lola Koala",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className=""
          onClick={[Function]}
        >
          View Bills
        </button>,
        <button
          className="active"
          onClick={[Function]}
        >
          View Chores
        </button>,
        null,
        <ChoresList
          claimChore={[Function]}
          currentUser={
            Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            }
          }
          markChoreDone={[Function]}
          placeRendered="/userlist"
          usersHouse={
            Object {
              "bills": Array [
                Object {
                  "allUsersTotals": Array [
                    Object {
                      "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                      "paid": false,
                      "total": 50,
                    },
                    Object {
                      "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                      "paid": false,
                      "total": 50,
                    },
                  ],
                  "datePosted": "11/01/2017",
                  "details": "We used a lot of water.",
                  "duedate": "11/9/17",
                  "id": "b3019762705779",
                  "parsedDuedate": 1510210800000,
                  "postedBy": Object {
                    "house": "Dog9",
                    "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    "name": "Amanda Brenner",
                  },
                  "title": "Water Bill",
                  "total": "100",
                },
              ],
              "bulletins": Array [
                Object {
                  "datePosted": "11/01/2017",
                  "details": "The dogs got out through a hole in the fence yesterday.",
                  "hasRead": Array [
                    "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                  ],
                  "id": 1509567098747,
                  "postedBy": Object {
                    "house": "Dog9",
                    "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    "name": "Amanda Brenner",
                  },
                  "title": "Don't leave dogs out unattended.",
                },
              ],
              "chores": Array [
                Object {
                  "assignedTo": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                  "datePosted": "10/31/2017",
                  "details": "Empty the dishwasher.",
                  "done": false,
                  "id": "c1509485241076",
                  "postedBy": Object {
                    "house": "Dog9",
                    "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    "name": "Amanda Brenner",
                  },
                  "title": "Dishes",
                  "urgency": "low",
                },
              ],
              "houseCode": "woof9",
              "houseKey": "-KxoIBWy5UYRscFYdWty",
              "houseName": "Dog9",
              "users": Array [
                Object {
                  "house": "Dog9",
                  "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                  "name": "Amanda Brenner",
                },
                Object {
                  "house": "Dog9",
                  "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                  "name": "Lola Koala",
                },
              ],
            }
          }
        />,
      ],
      "className": "user-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "View Bills",
          "className": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "View Bills",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "View Chores",
          "className": "active",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "View Chores",
        "type": "button",
      },
      null,
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "claimChore": [Function],
          "currentUser": Object {
            "house": "Dog9",
            "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
            "name": "Amanda Brenner",
          },
          "markChoreDone": [Function],
          "placeRendered": "/userlist",
          "usersHouse": Object {
            "bills": Array [
              Object {
                "allUsersTotals": Array [
                  Object {
                    "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    "paid": false,
                    "total": 50,
                  },
                  Object {
                    "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                    "paid": false,
                    "total": 50,
                  },
                ],
                "datePosted": "11/01/2017",
                "details": "We used a lot of water.",
                "duedate": "11/9/17",
                "id": "b3019762705779",
                "parsedDuedate": 1510210800000,
                "postedBy": Object {
                  "house": "Dog9",
                  "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                  "name": "Amanda Brenner",
                },
                "title": "Water Bill",
                "total": "100",
              },
            ],
            "bulletins": Array [
              Object {
                "datePosted": "11/01/2017",
                "details": "The dogs got out through a hole in the fence yesterday.",
                "hasRead": Array [
                  "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                ],
                "id": 1509567098747,
                "postedBy": Object {
                  "house": "Dog9",
                  "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                  "name": "Amanda Brenner",
                },
                "title": "Don't leave dogs out unattended.",
              },
            ],
            "chores": Array [
              Object {
                "assignedTo": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                "datePosted": "10/31/2017",
                "details": "Empty the dishwasher.",
                "done": false,
                "id": "c1509485241076",
                "postedBy": Object {
                  "house": "Dog9",
                  "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                  "name": "Amanda Brenner",
                },
                "title": "Dishes",
                "urgency": "low",
              },
            ],
            "houseCode": "woof9",
            "houseKey": "-KxoIBWy5UYRscFYdWty",
            "houseName": "Dog9",
            "users": Array [
              Object {
                "house": "Dog9",
                "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                "name": "Amanda Brenner",
              },
              Object {
                "house": "Dog9",
                "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                "name": "Lola Koala",
              },
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className=""
            onClick={[Function]}
          >
            View Bills
          </button>,
          <button
            className="active"
            onClick={[Function]}
          >
            View Chores
          </button>,
          null,
          <ChoresList
            claimChore={[Function]}
            currentUser={
              Object {
                "house": "Dog9",
                "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                "name": "Amanda Brenner",
              }
            }
            markChoreDone={[Function]}
            placeRendered="/userlist"
            usersHouse={
              Object {
                "bills": Array [
                  Object {
                    "allUsersTotals": Array [
                      Object {
                        "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                        "paid": false,
                        "total": 50,
                      },
                      Object {
                        "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                        "paid": false,
                        "total": 50,
                      },
                    ],
                    "datePosted": "11/01/2017",
                    "details": "We used a lot of water.",
                    "duedate": "11/9/17",
                    "id": "b3019762705779",
                    "parsedDuedate": 1510210800000,
                    "postedBy": Object {
                      "house": "Dog9",
                      "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                      "name": "Amanda Brenner",
                    },
                    "title": "Water Bill",
                    "total": "100",
                  },
                ],
                "bulletins": Array [
                  Object {
                    "datePosted": "11/01/2017",
                    "details": "The dogs got out through a hole in the fence yesterday.",
                    "hasRead": Array [
                      "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    ],
                    "id": 1509567098747,
                    "postedBy": Object {
                      "house": "Dog9",
                      "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                      "name": "Amanda Brenner",
                    },
                    "title": "Don't leave dogs out unattended.",
                  },
                ],
                "chores": Array [
                  Object {
                    "assignedTo": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    "datePosted": "10/31/2017",
                    "details": "Empty the dishwasher.",
                    "done": false,
                    "id": "c1509485241076",
                    "postedBy": Object {
                      "house": "Dog9",
                      "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                      "name": "Amanda Brenner",
                    },
                    "title": "Dishes",
                    "urgency": "low",
                  },
                ],
                "houseCode": "woof9",
                "houseKey": "-KxoIBWy5UYRscFYdWty",
                "houseName": "Dog9",
                "users": Array [
                  Object {
                    "house": "Dog9",
                    "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    "name": "Amanda Brenner",
                  },
                  Object {
                    "house": "Dog9",
                    "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                    "name": "Lola Koala",
                  },
                ],
              }
            }
          />,
        ],
        "className": "user-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "View Bills",
            "className": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "View Bills",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "View Chores",
            "className": "active",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "View Chores",
          "type": "button",
        },
        null,
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "claimChore": [Function],
            "currentUser": Object {
              "house": "Dog9",
              "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
              "name": "Amanda Brenner",
            },
            "markChoreDone": [Function],
            "placeRendered": "/userlist",
            "usersHouse": Object {
              "bills": Array [
                Object {
                  "allUsersTotals": Array [
                    Object {
                      "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                      "paid": false,
                      "total": 50,
                    },
                    Object {
                      "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                      "paid": false,
                      "total": 50,
                    },
                  ],
                  "datePosted": "11/01/2017",
                  "details": "We used a lot of water.",
                  "duedate": "11/9/17",
                  "id": "b3019762705779",
                  "parsedDuedate": 1510210800000,
                  "postedBy": Object {
                    "house": "Dog9",
                    "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    "name": "Amanda Brenner",
                  },
                  "title": "Water Bill",
                  "total": "100",
                },
              ],
              "bulletins": Array [
                Object {
                  "datePosted": "11/01/2017",
                  "details": "The dogs got out through a hole in the fence yesterday.",
                  "hasRead": Array [
                    "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                  ],
                  "id": 1509567098747,
                  "postedBy": Object {
                    "house": "Dog9",
                    "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    "name": "Amanda Brenner",
                  },
                  "title": "Don't leave dogs out unattended.",
                },
              ],
              "chores": Array [
                Object {
                  "assignedTo": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                  "datePosted": "10/31/2017",
                  "details": "Empty the dishwasher.",
                  "done": false,
                  "id": "c1509485241076",
                  "postedBy": Object {
                    "house": "Dog9",
                    "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                    "name": "Amanda Brenner",
                  },
                  "title": "Dishes",
                  "urgency": "low",
                },
              ],
              "houseCode": "woof9",
              "houseKey": "-KxoIBWy5UYRscFYdWty",
              "houseName": "Dog9",
              "users": Array [
                Object {
                  "house": "Dog9",
                  "id": "RVde1m9BhkQBLZlTJNsFlQZXqP43",
                  "name": "Amanda Brenner",
                },
                Object {
                  "house": "Dog9",
                  "id": "cFNdorl5ceRyMr9XGvPFRVEcw7I3",
                  "name": "Lola Koala",
                },
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
